lib/signaling/participant.js

```diff
@@ -198,14 +208,30 @@ class ParticipantSignaling extends StateMachine {
    * @returns {boolean}
    */
   connect(sid, identity) {
-    if (this.state === 'connecting') {
-      this._sid = sid;
-      this._identity = identity;
+    if (this.state === 'connecting' || this.state === 'reconnecting') {
+      if (!this._sid) {
+        this._sid = sid;
+      }
+      if (!this._identity) {
+        this._identity = identity;
+      }
       this.preempt('connected');
       return true;
     }
     return false;
   }
+
+  /**
+   * Transition to "reconnecting" state.
+   * @returns {boolean}
+   */
+  reconnecting() {
+    if (this.state === 'connecting' || this.state === 'connected') {
+      this.preempt('reconnecting');
+      return true;
+    }
+    return false;
```

Comment:
curious - why is this check necessary - in what cases would we get here ?

Reply:
@makarandp0 Transitions that are not permitted will result in an exception. Hence the check.

Reply:
in that case, same point - we should log error for unexpected cases, since we are handling it anyways.